app/bundles/EmailBundle/Tests/Controller/PublicControllerFunctionalTest.php

```diff
@@ -31,6 +34,20 @@ public function testUnsubscribeFormActionWithoutTheme(): void
         $this->assertTrue($this->client->getResponse()->isOk());
     }
 
+    public function testContactPreferencesSaveMessage(): void
+    {
+        $lead = $this->getLead();
+        $stat = $this->getStat(null, $lead);
+        $this->em->flush();
+
+        $crawler = $this->client->request('GET', '/email/unsubscribe/'.$stat->getTrackingHash());
+        $this->assertStringContainsString('/email/unsubscribe/tracking_hash_unsubscribe_form_email', $crawler->filter('form')->eq(0)->attr('action'));
+        $crawler = $this->client->submitForm('Save');
+
+        $this->assertEquals(1, $crawler->filter('#success-message-text')->count());
```

Comment:
Can we assert also what the actual text is? That's the point of this feature, right?
```suggestion
        $this->assertEquals(1, $crawler->filter('#success-message-text')->count());
        $this->assertEquals('Success', $crawler->filter('#success-message-text')->text());
```